### PR TITLE
Add "Open All" button to blocked mods dialog

### LINKED ISSUE #820: add an "open all" button for opted out mod page
A small fix that improves a lot UX

### PR DESCRIPTION
**Pull request by kumquat-ir**
Fixes #820

I'm not the best at C++, could probably be improved

Has some weird behavior for me, it opens one tab first and the rest a few seconds later
Not sure why, or how to fix

Renames classes since they were only being used for this dialog anyways

### CODE REVIEW

**Comment by DioEgizio** [comment]
> Renames classes since they were only being used for this dialog anyways

#588

**Comment by kumquat-ir** [comment]
*sigh*
i really should have thought of that honestly

**Comment by DioEgizio** [comment]
even if the behaviour is a bit weird (like you said), seems to work and it's a nice UX improvement. thanks

**Review by kthchew — APPROVED** [review]
Seems to work great!

> Has some weird behavior for me, it opens one tab first and the rest a few seconds later
Not sure why, or how to fix

I don't see this issue on macOS when opening 3 URLs at once. Maybe it's an OS-specific Qt bug, unless it only happens when there's more URLs to open?

**Comment by kumquat-ir** [comment]
happened to me with 3 urls on linux, probably os- or setup-specific in that case

**Comment by DioEgizio** [comment]
> I don't see this issue on macOS when opening 3 URLs at once. Maybe it's an OS-specific Qt bug, unless it only happens when there's more URLs to open?

i can reproduce that on windows too though

File: launcher/ui/dialogs/BlockedModsDialog.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#include "BlockedModsDialog.h"
+#include "ui_BlockedModsDialog.h"
+#include "qpushbutton.h"
```
Comment (flowln) [inline launcher/ui/dialogs/BlockedModsDialog.cpp]:
```suggestion
#include <QPushButton>
```

**Review by flowln — APPROVED** [review]
ehm, these blocked and scroll message dialogs are getting out of hand, i should stop delaying it and refactor those once and for all xd

this looks fine! the delay with multiple urls also happens on my end, but it's no big deal.

**Comment by Scrumplex** [comment]
The resolver for FTB modpacks also uses a similar dialog. Can you update that as well? 
https://github.com/PolyMC/PolyMC/blob/b15544c163ccbca08ea498b3b3a51c82d7fb3e12/launcher/modplatform/modpacksch/FTBPackInstallTask.cpp#L201-L215

You probably need to rebase onto develop / merge it into your branch first. It was introduced by #699

**Review by Scrumplex — APPROVED** [review]
Thanks!

**Pull request merged**